### PR TITLE
feat(package-cache): add configurable cache TTL override

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
Adds a new global config option which overrides default TTL for set namespace.

## Context
- https://github.com/renovatebot/renovate/discussions/23806

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -2359,6 +2359,19 @@ const options: RenovateOptions[] = [
...
   },
+  {
+    name: 'cacheTtlOverride',
+    description: 'An object that contains cache namespace TTL override values',
+    type: 'object',
+    stage: 'repository',
+    default: {},
+    additionalProperties: {
+      type: 'number',
+    },
```
Comment (Gabriel-Ladzaretti) [inline lib/config/options/index.ts:2370]: not sure I'm using this field as intended. 
my assumption is that it corresponds to the Record key type.

Comment (rarkins) [reply lib/config/options/index.ts:2370]: Check the generated JSON schema to see if it validates how you expect

File: lib/util/cache/package/decorator.ts [hunk 2]
```diff
@@ -76,7 +76,8 @@ export function cache<T>({
       finalKey
     );
 
-    const softTtl = ttlMinutes;
+    const ttlOverride = getTtlOverride(namespace);
```
Comment (viceice) [inline lib/util/cache/package/decorator.ts:79]:
```suggestion
    const ttlOverride = getTtlOverride(finalNamespace);
```
so we ensure always string?

File: lib/config/options/index.ts (hunk 1 above)
Comment (Gabriel-Ladzaretti) [reply lib/config/options/index.ts]: it doesnt provide any additional type safety, and the schema looks the same.
so i've removed it.

File: docs/usage/self-hosted-configuration.md [hunk 3]
```diff
@@ -207,6 +207,20 @@ Results which are soft expired are reused in the following manner:
 - The `etag` from the cached results will be reused, and may result in a 304 response, meaning cached results are revalidated
 - If an error occurs when querying the `npmjs` registry, then soft expired results will be reused if they are present
 
+## cacheTtlOverride
+
+Utilize this key-value map to override the default package cache TTL values for a specific namespace. This object contains pairs of namespaces and their corresponding TTL values in minutes.
+For instance, to override the default TTL of 60 minutes for the Docker datasource "tags"
+namespace: `datasource-docker-tags` use the following:
```
Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:214]:
```suggestion
For example, to override the default TTL of 60 minutes for the `docker` datasource "tags" namespace: `datasource-docker-tags` use the following:
```

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**